ambition_utils/rrule/models.py

```diff
@@ -304,3 +304,32 @@ def generate_dates_from_params(cls, rrule_params, time_zone=None, num_dates=20):
         rule = cls(rrule_params=rrule_params, time_zone=time_zone)
 
         return rule.generate_dates(num_dates=num_dates)
+
+    @classmethod
+    def clone_with_offset(cls, source: RRule, day_offset: int) -> RRule:
+        """
+        Creates a clone of a passed RRule object offset by a specified number of days.
+        Days can be negative.
+        """
+
+        # Create a clone of the source RRule
+        clone = copy.deepcopy(source)
+
+        # Get rrule object
+
+        # Update the rrule.dtstart with the offset.
+        clone.rrule_params['dtstart'] = parser.parse(clone.rrule_params['dtstart']) + timedelta(days=day_offset)
+
+        # Update the next_occurrence with the calculated offset
+        # refresh_occurrence() is not what we want because the clone needs an explict next_occurrence.
+        clone.next_occurrence = clone.rrule_params['dtstart']
```

Comment:
_From [#10563](https://github.com/ambitioninc/ambition/pull/10563)_

> recurrence.next_occurrence = time_to_schedule_first_meeting
if you are changing the recurrence config, you should be able to call a method on that class to update the next occurrence.

I'm assuming you are referring to `refresh_next_occurrence()`. I attempted to use it but it always returned the second date. I wonder if passing a `current_time` in the past would generate that first date I seek. Even so, that feels messy.